Source/VMware.PSDesiredStateConfiguration/Classes/DscItems.ps1

```diff
@@ -147,3 +964,211 @@ class DscTestMethodDetailedResult {
     }
 }
 
+<#
+.DESCRIPTION
+Executes Configuration objects created from New-VmwDscConfiguration cmdlet.
+#>
+class DscConfigurationRunner {
+    [VmwDscConfiguration] $Configuration
+    [string] $DscMethod
+
+    hidden [VmwDscNode[]] $ValidNodes
+
+    DscConfigurationRunner([VmwDscConfiguration] $Configuration, [string] $DscMethod) {
+        $this.Configuration = $Configuration
+        $this.DscMethod = $DscMethod
+    }
+
+    <#
+    .DESCRIPTION
+    Invokes the configuration
+    #>
+    [Psobject] InvokeConfiguration() {
+        $this.ValidateVsphereNodes()
+
+        $invokeResult = New-Object 'System.Collections.ArrayList'
+
+        foreach ($node in $this.ValidNodes) {
+            Write-Progress "Invoking node with name: $($node.InstanceName)"
+            
+            $result = $this.InvokeNodeResources($node.Resources)
+
+            $nodeResult = [PsObject]@{
+                OriginalNode = $node
+                InvokeResult = $result
+            }
+
+            $invokeResult.Add($nodeResult) | Out-Null
+        }
+
+        return $invokeResult
+    }
+
+    <#
+    .DESCRIPTION
+    Checks if there are any vSphere Nodes and validates them.
+    Sets ValidNodes property for use in invoking the nodes
+    #>
+    hidden [void] ValidateVsphereNodes() {
+        $vSphereNodes = $this.Configuration.Nodes | Where-Object { $_ -is [VmwVsphereDscNode] }
+
+        if ($null -eq $vSphereNodes -or $vSphereNodes.Count -eq 0) {
+            $this.ValidNodes = $this.Configuration.Nodes
+
+            return
+        }
+
+        # gets PSVersionTable via Get-Variable because automatic variables are not accessable in any other way inside classes
+        $psVersionTableVariable = Get-Variable -Name PSVersionTable
+
+        if ($psVersionTableVariable.Value['PSEdition'] -ne 'Core') {
+            throw $Script:VsphereNodesAreOnlySupportedOnPowerShellCoreException
+        }
+
+        $validVsphereNodes = New-Object 'System.Collections.ArrayList'
+
+        $defaultViServers = $this.GetDefaultViServers()
```

Comment:
the name of the variable matches the PowerCLI $global:defaultVIServers. Are you sure you are not overriding the global variable this way?